src/DaprExtension/Bindings/DaprSecretConverter.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions.Dapr
+{
+    using System;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.WebJobs;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+
+    class DaprSecretConverter :
+        IAsyncConverter<DaprSecretAttribute, JObject>,
+        IAsyncConverter<DaprSecretAttribute, string?>,
+        IAsyncConverter<DaprSecretAttribute, byte[]>
+    {
+        readonly DaprServiceClient daprClient;
+
+        public DaprSecretConverter(DaprServiceClient daprClient)
+        {
+            this.daprClient = daprClient;
+        }
+
+        Task<JObject> IAsyncConverter<DaprSecretAttribute, JObject>.ConvertAsync(
+            DaprSecretAttribute input,
+            CancellationToken cancellationToken)
+        {
+            return this.GetSecretsAsync(input, cancellationToken);
+        }
+
+        async Task<string?> IAsyncConverter<DaprSecretAttribute, string?>.ConvertAsync(
+            DaprSecretAttribute input,
+            CancellationToken cancellationToken)
+        {
+            JObject result = await this.GetSecretsAsync(input, cancellationToken);
+            return result.PropertyValues().FirstOrDefault()?.ToString();
```

Comment:
Can bindings evaluate to `null`? I vaguely remember something about WebJobs throwing an error if a binding parameter is null, but I could be misremembering here.

Reply:
Also, even if we can evaluate to null here, I'm not sure it is the right option? In the other two types, the value when the secret doesn't exist is the empty version of that type. I'm not sure if it is more important to be more consistent binding-wide across the types, or more important to be consistent with how users expect the type to behave.

Reply:
Yeah, I'm not being consistent in my null handling between string and `byte[]`. I originally went with null but then ran into a problem where the StyleCop version I'm using doesn't understand that `byte[]?` as valid syntax. I "fixed" the warning by using an empty byte array instead. I don't have a good sense for how important it is to be consistent across different types.

And yes, I'm pretty sure bindings can evaluate to null as long as the object type you're binding to is a reference type.